services/submission/src/main/java/app/coronawarn/server/services/submission/validation/ValidSubmissionPayload.java

```diff
@@ -132,16 +136,45 @@ private boolean checkNoOverlapsInTimeWindow(List<TemporaryExposureKey> exposureK
 
       Integer[] sortedStartIntervalNumbers = exposureKeys.stream()
           .mapToInt(TemporaryExposureKey::getRollingStartIntervalNumber)
-          .sorted().boxed().toArray(Integer[]::new);
+          .distinct().sorted().boxed().toArray(Integer[]::new);
 
       for (int i = 1; i < sortedStartIntervalNumbers.length; i++) {
-        if ((sortedStartIntervalNumbers[i - 1] + EXPECTED_ROLLING_PERIOD) > sortedStartIntervalNumbers[i]) {
+        if ((sortedStartIntervalNumbers[i - 1] + maxRollingPeriod) > sortedStartIntervalNumbers[i]) {
           addViolation(validatorContext, String.format(
               "Subsequent intervals overlap. StartIntervalNumbers: %s", sortedStartIntervalNumbers));
           return false;
         }
       }
       return true;
     }
+
+    private boolean checkKeysCummulateEqualOrLessThanMaxRollingPeriodPerDay(List<TemporaryExposureKey> exposureKeys,
+        ConstraintValidatorContext validatorContext) {
+
+      HashMap<Integer, Integer> totalKeysPerDay = new HashMap<>();
+
+      for (TemporaryExposureKey exposureKey : exposureKeys) {
+        int numberOfKeys = exposureKey.getRollingPeriod();
+        if (totalKeysPerDay.containsKey(exposureKey.getRollingStartIntervalNumber())) {
+          numberOfKeys += totalKeysPerDay.get(exposureKey.getRollingStartIntervalNumber());
```

Comment:
I'm not sure if this is testing the correct constraint. AFAIK, the rolling period sums of all TEKs with the same rolling start interval number must be <= 144. But here, the number of keys are being counted?

Reply:
Here the name of the variables are misleading. I will rename them